src/run/run_step.go

```diff
@@ -46,7 +70,16 @@ func Parallel(ctx context.Context, state *core.BuildState, labels []core.Annotat
 		g.Go(func() error {
 			limiter <- struct{}{}
 			defer func() { <-limiter }()
-			return run(state, label, args, true, quiet, remote, env, detach, inTmp, dir)
+
+			quiet := output == Quiet || output == GroupImmediate
+			out, err := run(state, label, args, true, quiet, remote, env, detach, inTmp, dir)
+
+			if output == GroupImmediate {
+				fmt.Println(label)
```

Comment:
This will only prefix the label at the start of the whole output, and will wait for it to complete before printing anything.

I think you might want to wrap the `Stdout` and `Stderr` of the command with a writer that prefixes them?

Reply:
At the moment it is printing the label and output at the same time, and only after the task finishes.

Having the following:
```python
sh_cmd(
    name = "one",
    cmd = [
        "echo one",
        "sleep 3",
    ],
)

sh_cmd(
    name = "two",
    cmd = [
        "echo two",
        "sleep 2"
    ],
)
```
And running `plz run //src:please -- run parallel //docs:one //docs:two --output=group_immediate`

Outputs the following after 2s
```
//docs:two
two
```

and then the following after 1s (2s from the previous)
```
//docs:one
one
```

Unless I'm missing something?

Reply:
@pebers I think VJ wanted some way to group the output as a priority. The prefix idea was just something I thought of, but wasn't quite what was requested. We talked about printing the prefixed stuff during execution that so we get some partial output before clearing the output and printing this all this. I'd rather get on to the VS code debugging though.

Reply:
Yes, this does the grouping but not the prefixing on every line. The reason for this was the possibility of long build label prefixes during discussion. Sticking it to the top would help avoid that problem and still help differentiate the output. I guess we could make the output top-line label more pronounced?

Reply:
Oh OK sorry, I was just thinking of something different then.